tutorials/shading/visual_shader.rst

```diff
@@ -0,0 +1,101 @@
+.. _doc_visual_shader:
+
+Visual Shader
+=============
+
+Introduction
+------------
+
+VisualScript is an alternative for users that cannot code, Similarly, Visual Shader is the visual alternative for creating shaders.
+
+.. note::
+    
+   If you do not understand what shader is , go read :ref:`doc_what_are_shaders`
+   
+Creating a VisualShader
+-----------------------
+
+Create a new ``ShaderMaterial`` in some object of your choice.
+
+.. image:: img/shader_material_create.png
+
+Create a new ``VisualShader`` inside the ``ShaderMaterial`` and the visual shader editor will open automatically.
+
+.. image:: img/shader_create.png
+
+The layout of the visual shader editor can be divided into two parts the upper bar and the graph.
+
+.. image:: img/visual_shader_editor2.png
+
+From left to right of the upper bar, the ``Add Node`` button allows you to add more nodes to the shader graph, to the right, is the shader mode : ``vertex``, ``fragment`` and ``light``. Then, it is the zooming level and the snapping and grid options. The last icon can show the resulting shader code that the graph creates.
+
+Using the Visual Shader Editor
+------------------------------
+
+By default, every new ``VisualShader`` will have an output node. The node connection usually ends at one of the output socket. A node is the basic unit to create your shader. To add a new node, click on the ``Add Node`` button on the upper left corner or right click on any empty space inside the Visual Shader Editor, a menu will pop up.
```

Comment:
"Every node connection ends at one of the output node's sockets."

"...space inside the Visual Shader Editor, and a menu will pop up."